charts/core-dump-handler/templates/secrets.yaml

```diff
@@ -14,3 +14,11 @@ stringData:
   s3BucketName: {{ .Values.daemonset.s3BucketName }}
   s3Region: {{ .Values.daemonset.s3Region }}
 {{- end }}
+---
+apiVersion: v1
```

Comment:
Can this be put behind a flag such as 
```yaml
{{- if .Values.daemonset.useServiceAccountToken }}
apiVersion: v1
.....
```
With the value set to false in the `values.yaml` 
https://github.com/IBM/core-dump-handler/blob/main/charts/core-dump-handler/values.yaml#L60
```yaml
useServiceAccountToken: false
```
If this is mandatory now for all AWS instances can you add `useServiceAccountToken: false`
And the value set to true in `aws.values.yaml`?

Reply:
Thanks for your feedback! I've moved the secret to a dedicated file and added the flag, but to the `serviceAccount` section, seemed more intuitive to me.

Also want to outline, that this does not seem to be limited to AWS (https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.24.md#no-really-you-must-read-this-before-you-upgrade-1). I can't test/verify this for any other environment, have no access to any. But if it turns out it is requires, I could add the respective config to the other values files.